proto/osmosis/tokenfactory/v1beta1/genesis.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+package osmosis.tokenfactory.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "osmosis/tokenfactory/v1beta1/authorityMetadata.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/x/tokenfactory/types";
+
+// GenesisState defines the tokenfactory module's genesis state.
+message GenesisState {
+  option (gogoproto.equal) = true;
+
+  repeated GenesisDenom factory_denoms = 1 [
+    (gogoproto.moretags) = "yaml:\"factory_denoms\"",
+    (gogoproto.nullable) = false
+  ];
+}
+
+message GenesisDenom {
```

Comment:
Wdym?  This is just a struct that pairs a denom with its authority metadata, so that it can be included in a genesis file.  I don't think this merged struct is used anywhere else.

Reply:
Ah yeah, was looking at this at the beginning of review.

In state, it stores authority_metadata at a key for the denom. I was thinking we would be storing the entire struct as the value for simplicity, rn we just store Authority_metadata. (Which is more efficient, just means we have to deal with a separate genesis type)

I can't wait for the ORM module, kinda ridiculous we have to basically maintain our own indexing by every entry. (Ideally we'd just have a single struct, and just specify what we want to index on, ORM does this)

This is good